libs/promises/api/celix/DefaultScheduledExecutor.h

```diff
@@ -0,0 +1,138 @@
+/**
+ *Licensed to the Apache Software Foundation (ASF) under one
+ *or more contributor license agreements.  See the NOTICE file
+ *distributed with this work for additional information
+ *regarding copyright ownership.  The ASF licenses this file
+ *to you under the Apache License, Version 2.0 (the
+ *"License"); you may not use this file except in compliance
+ *with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *Unless required by applicable law or agreed to in writing,
+ *software distributed under the License is distributed on an
+ *"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ *specific language governing permissions and limitations
+ *under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <future>
+#include <functional>
+#include <unordered_map>
+
+#include "celix/IScheduledExecutor.h"
+
+namespace celix {
+
+    class DefaultDelayedScheduledFuture : public celix::IScheduledFuture {
+    public:
+        explicit DefaultDelayedScheduledFuture(std::chrono::duration<double, std::milli> _delayInMs) : delayInMs{_delayInMs} {}
+
+        ~DefaultDelayedScheduledFuture() noexcept override = default;
+
+        bool isCancelled() const override {
+            std::lock_guard<std::mutex> lock{mutex};
```

Comment:
In C++17, specifying the template type is not necessary here. More info [here](https://accu.org/journals/overload/26/143/orr_2465/).

Reply:
Coo, was not aware of this yet. 

CTAD FTW ;), this can cleanup a lot of unnecessary type specifications.